test/expect-tests/dune_rpc_e2e/dune_rpc_diagnostics.ml

```diff
@@ -711,5 +711,86 @@ let g = A.f
           ; [ "related"; [] ]
           ; [ "targets"; [] ]
           ]
+        ]
+        [ "Add"
```

Comment:
Because they are separate errors. "related" errors are really for errors that have more than one location. The most common example is an .ml/.mli mismatch where there's a location in the .ml and the .mli inside the error message.